File: PrivateD2party/solve/java/PrivateD2partySolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class PrivateD2partySolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        List<Integer> aBoxed = (List<Integer>) reader.next(new TypeRef<List<Integer>>(){}.getType());
        int[] a = new int[aBoxed.size()];
        for (int _i = 0; _i < aBoxed.size(); ++_i)
            a[_i] = aBoxed.get(_i);
        reader.close();

        PrivateD2party solver = new PrivateD2party();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.getsz(a));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }    
}

File: PrivateD2party/solve/cpp/PrivateD2partySolver.cpp
#include <fstream>
#include <topcoder>
#include "PrivateD2party.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
    try {
        std::ifstream ifs(argv[1]);
        vector<int> a; tc::read(ifs, a);
        ifs.close();

        std::ofstream ofs(argv[2]);
        PrivateD2party solver;
        tc::show(ofs, solver.getsz(a));
        ofs.close();
    } catch (std::exception &e) {
        std::cerr << e.what() << std::endl;
    }
    return 0;
}